server/protocols/jmap/src/main/java/org/apache/james/jmap/model/VacationResponse.java

```diff
@@ -114,24 +109,21 @@ public Builder fromVacation(Vacation vacation, ZonedDateTime zonedDateTime) {
         }
 
         public VacationResponse build() {
-            boolean enabled = isEnabled.orElse(DEFAULT_DISABLED);
-            if (enabled) {
-                Preconditions.checkState(textBody.isPresent() || htmlBody.isPresent(), "textBody or htmlBody property of vacationResponse object should not be null when enabled");
-            }
-            return new VacationResponse(id, enabled, fromDate, toDate, textBody, subject, htmlBody);
+            Preconditions.checkState(!isEnabled.isNull());
+            return new VacationResponse(id, isEnabled, fromDate, toDate, textBody, subject, htmlBody);
         }
     }
 
-    private final String id;
-    private final boolean isEnabled;
-    private final Optional<ZonedDateTime> fromDate;
-    private final Optional<ZonedDateTime> toDate;
-    private final Optional<String> subject;
-    private final Optional<String> textBody;
-    private final Optional<String> htmlBody;
+    private final MissableOptional<String> id;
```

Comment:
When updating the vacation, it is stated in the spec one can ommit any field

> The object may omit any property; only properties that have changed need be included.

So yes, id is not always mandatory, as it is a property always set to "singleton". According to the spec, there is no ambiguity here.